core/templates/pages/topics-and-skills-dashboard-page/modals/delete-skill-modal.component.ts

```diff
@@ -46,8 +47,24 @@ export class DeleteSkillModalComponent extends ConfirmOrCancelModal {
     this.topicsAndSkillsDashboardBackendApiService
       .fetchTopicAssignmentsForSkillAsync(
         this.skillId).then((response: AssignedSkill[]) => {
-        this.topicsAssignments = response;
-        this.topicsAssignmentsAreFetched = true;
+        let allTopicIds = [];
+        for (let topic of response) {
+          allTopicIds.push(topic.topicId);
+        }
+        this.topicsAndSkillsDashboardBackendApiService
+          .fetchTopicIdToDiagnosticTestSkillIdsAsync(allTopicIds).then(
+            (dict) => {
+              for (let topicId in dict.topicIdToDiagnosticTestSkillIds) {
+                let diagnosticTestSkillIds = (
+                  dict.topicIdToDiagnosticTestSkillIds[topicId]);
+                if (diagnosticTestSkillIds.length === 1 &&
+                    diagnosticTestSkillIds.indexOf(this.skillId) !== -1) {
+                  this.allowSkillForDeletion = false;
```

Comment:
Perhaps allowDeletionOfSkill (if that's what you mean)?

Reply:
Updated